adapter/stackdriver/log/log.go

```diff
@@ -131,13 +132,22 @@ func (h *handler) HandleLogEntry(_ context.Context, values []*logentry.Instance)
 		payload := buf.String()
 		pool.PutBuffer(buf)
 
-		linfo.log(logging.Entry{
+		e := logging.Entry{
 			Timestamp:   h.now(), // TODO: use timestamp on Instance when timestamps work
 			Severity:    logging.ParseSeverity(v.Severity),
 			Labels:      toLabelMap(linfo.labels, v.Variables),
 			Payload:     payload,
 			HTTPRequest: toReq(linfo.req, v.Variables),
-		})
+		}
+
+		// If we don't set a resource the SDK will populate a global resource for us.
+		if v.MonitoredResourceType != "" {
+			e.Resource = &monitoredres.MonitoredResource{
+				Type:   v.MonitoredResourceType,
+				Labels: helper.ToStringMap(v.MonitoredResourceDimensions),
```

Comment:
I think if we make the template for MonitoredResourceDimenstions as map<string, string> you don't need the ToStringMap

Reply:
I intentionally didn't so that the operator could use any expressions they wanted, and we handle making them into strings since that's what the API requires. For example there's no IP address to string function in our expression language, so they could not use IP_ADDRESS expressions as labels, which is a pain.

Reply:
Got it.. so if it was map<string, string>, operator won't be able to pass IP_ADDRESS expression.

Reply:
Exactly.